configure.ac

```diff
@@ -167,28 +167,86 @@ AM_CONDITIONAL(WNCKLET_INPROCESS, test -n "$WNCKLET_COMPILE_INPROCESS")
 # For the run dialog
 gl_CHECK_TYPE_STRUCT_DIRENT_D_TYPE
 
-dnl X development libraries check
-
-PKG_CHECK_MODULES(X, x11 xau, :, [
-  # pkg-config modules not found (only present since X11R7 aka Xorg); use
-  # old-style detection
-  AC_PATH_XTRA()
-  # X not found
-  if test x$no_x = xyes ; then
-    AC_MSG_ERROR([X development libraries not found])
-  fi
-
-  gp_save_cflags="$CFLAGS"
-  gp_save_libs="$LIBS"
-  CFLAGS="$X_CFLAGS"
-  LIBS="$X_PRE_LIBS $X_LIBS $X_EXTRA_LIBS"
-  AC_CHECK_LIB(X11, XFree,, AC_MSG_ERROR([libX11 not found]))
-  AC_CHECK_LIB(Xau, XauFileName,, AC_MSG_ERROR([libXau not found]))
-  CFLAGS="$gp_save_cflags"
-  LIBS="$gp_save_libs"
-
-  X_LIBS="$X_PRE_LIBS $X_LIBS -lX11 -lXau $X_EXTRA_LIBS"
-])
+must_enable_x11=no
+must_enable_wayland=no
+try_all_backends=yes
+
+AC_ARG_ENABLE(x11,
+              [AC_HELP_STRING([--enable-x11],
+                              [Enable X11 support and disable Wayland (unless it's also enabled explicitly)
+                              @<:@default=yes@:>@])],
+              [must_enable_x11=yes
+               try_all_backends=no],)
+
+AC_ARG_ENABLE(wayland,
+              [AC_HELP_STRING([--enable-wayland],
+                              [Enable Wayland support and disable X11 (unless it's also enabled explicitly)
+                              @<:@default=yes@:>@])],
+              [must_enable_wayland=yes
+               try_all_backends=no],)
+
+# Check if we have wayland-client installed, and thus should build with Wayland support
+
+have_wayland=no
+
+if test "x$must_enable_wayland" = "xyes" -o "x$try_all_backends" = "xyes"; then
+  PKG_CHECK_MODULES(WAYLAND, wayland-client, have_wayland=yes,)
+fi
+
+AM_CONDITIONAL(ENABLE_WAYLAND, [test "x$have_wayland" = "xyes"])
+
+if test "x$have_wayland" = "xyes"; then
+  AC_DEFINE(HAVE_WAYLAND, 1, [Have the Wayland development library])
+fi
+
+# Error out if Wayland was required but not found
+if test "x$have_wayland" != "xyes" -a "x$must_enable_wayland" = "xyes"; then
+  # The user has explicitly told us to use Wayland, but we can't find the library :(
+  AC_MSG_ERROR([Wayland client library not found])
+fi
+
+# Check if we have the X development libraries
+# (we can compile with X and/or Wayland supported)
+
+have_x11=no
+
+if test "x$must_enable_x11" = "xyes" -o "x$try_all_backends" = "xyes"; then
+  PKG_CHECK_MODULES(X, x11 xau, have_x11=yes, [
+    # pkg-config modules not found (only present since X11R7 aka Xorg); use
+    # old-style detection
```

Comment:
Well it is 2018... I would just require pkg-config file and would remove old-style detection...

https://gitlab.freedesktop.org/xorg/lib/libx11
https://gitlab.freedesktop.org/xorg/lib/libxau
This tells that x11.pc and xau.pc is available as minimum for 9 years.